src/library/scala/collection/immutable/HashMap.scala

```diff
@@ -197,7 +197,7 @@ object HashMap extends ImmutableMapFactory[HashMap] with BitOperations.Int {
           if (this.value.asInstanceOf[AnyRef] eq value.asInstanceOf[AnyRef]) this
           else new HashMap1(key, hash, value, kv)
         } else {
-          val nkv = merger(this.kv, kv)
+          val nkv = merger(this.ensurePair, if(kv != null) kv else (key->value))
```

Comment:
since this is relatively low-level code, could you create the pair as `(key, value)` instead of using `->`?